### ISSUE TITLE
uuid.FromString("6ba7b8109dad11d180b400c04fd430c8") does not work, and really should.

### ISSUE THREAD

**Issue by duaneking** [satori/go.uuid#57]
I feel the tests have an error in them, in that a format is not supported that really, really should be.

While **uuid.FromString("6ba7b810-9dad-11d1-80b4-00c04fd430c8")** is clearly supported, the string **uuid.FromString(""6ba7b8109dad11d180b400c04fd430c8")** should also work as well as its considered a valid uuid in many languages and schema frameworks and so this support is important for interoperability... just it does not work at the moment even though it really should, and this has caused us (and I'm sure, many many others) all kinds of issues.

For example, this lack of support for UUID's without dashes makes dealing with marshmallow and Python generated .hex UUID values a big PITA when you trying to use the hex value as a primary key in a DB or massively distributed system.

It also means that while other libraries and languages are generating these "valid UUID's", the UUID framework itself does not support them, so it makes it look like this library doesn't have full support for UUID's.

I understand that some may want to create an overly pedantic discussion about if a UUID without a dash is really a UUID that won't take into account the fact real people are using these libraries to get work done and so this support would make lives easier, but the fact is most people consider them to be valid, many other frameworks consider them to be valid,  and so this support really should exist if the core golang philosophy of getting stuff done is to be respected.

**Comment by jlburkhead** [comment]
why not write a wrapper around `uuid.FromString`? 

```go
func uuidFromString(s string) (uuid.UUID, error) {
	if len(s) == len(uuid.Nil.String()) {
		return uuid.FromString(s)
	}
	b, err := hex.DecodeString(s)
	if err != nil {
		return uuid.Nil, err
	}
	return uuid.FromBytes(b)
}
```

**Comment by duaneking** [comment]
I thank you for your response, but wouldn't that wrapper be much slower than a more native implementation?

**Comment by ivanjaros** [comment]
To my knowledge the dashes are in the standard so not having them is non-standard and you should handle those cases on your own.

**Comment by daskol** [comment]
@duaneking Look at issue #54. I proposed there more efficient implementation than @jlburkhead suggested. Moreover if you do not want apply my patch by yourself you could use [fork](/edadeal/uuid) of @edadeal. We maintain and develop this project since @satori stop maintaince and even do not reply any comments.

**Comment by satori** [comment]
Closed by merging #54

**Comment by duaneking** [comment]
👍
